pylint/config/config_initialization.py

```diff
@@ -83,7 +83,17 @@ def _config_initialization(
         msg = ", ".join(unrecognized_options)
         linter._arg_parser.error(f"Unrecognized option found: {msg}")
 
-    # Set the current module to configuration as we don't know where
+    # Now that config file and command line options have been loaded
+    # with all disables, it is safe to emit messages
+    if unrecognized_options_message is not None:
+        linter.set_current_module(str(config_file) if config_file else "")
+        linter.add_message(
+            "unrecognized-option", args=unrecognized_options_message, line=0
+        )
+
+    linter._emit_bad_option_value()
+
+    # Set the current module to include command line as we don't know where
```

Comment:
I can revert back to the original comment after the latest edits, as it now makes sense again.
```suggestion
    # Set the current module to configuration as we don't know where
```